tools/gen_addon_icon.py

```diff
@@ -0,0 +1,96 @@
+# License AGPLv3 (http://www.gnu.org/licenses/agpl-3.0-standalone.html)
+# Copyright (c) 2019 Eficent Business and IT Consulting Services S.L.
+#        (http://www.eficent.com)
+
+import io
+import os
+
+import click
+
+from .gitutils import commit_if_needed
+from .manifest import read_manifest, find_addons, NoManifestFound
+
+
+ICONS_DIR = os.path.join('static', 'description')
+
+
+def gen_one_addon_icon(addon_dir):
+    # put
+    template_filename = \
+        os.path.join(os.path.dirname(__file__).rpartition('tools')[0],
+                     'template', 'module',
+                     ICONS_DIR, 'icon.svg')
+    icon_filename = \
+        os.path.join(
+            addon_dir, ICONS_DIR, 'icon.png',
+        )
+    with io.open(template_filename, 'rb') as tf:
+        icon = tf.read()
+    with io.open(icon_filename, 'wb') as rf:
+        rf.write(icon)
+    return icon_filename
+
+
+def gen_one_addon_svg(icon_filename):
+    icon_dir = os.path.dirname(icon_filename)
+    svg_filename = os.path.join(icon_dir, 'icon.svg')
+    if os.path.exists(svg_filename):
+        # svg was created manually
+        return
```

Comment:
I you test this here, you should do the same in gen_one_addon_icon too, for symmetry.

Reply:
well, I don't test this in `gen_one_addon_icon` because `gen_one_addon_icon` is only called in `gen_addon_icon` and it's already tested in `gen_addon_icon` before calling `gen_one_addon_icon`.

Reply:
Yes, that's what I say, it's assymetric and the code path is harder to follow. The could would be much simpler if both function test for the existence of the target filename.

Reply:
ah, ok